### DISCUSSION TITLE
Has the behaviour of Single() changed?

### DISCUSSION THREAD

**Discussion by kelvinpompey** [supabase/supabase#1354] [Questions] [answered]
I am using supabase-js 1.11.7 and it's returning an error when single() is used on an empty table. Previous I think it returned null. Is this a bug? 

```
{
  details: 'Results contain 0 rows, application/vnd.pgrst.object+json requires 1 row',
  message: 'JSON object requested, multiple (or no) rows returned'
}
```

**Comment by atnegrete** [comment, marked as answer]
If it returned null previously, I'm guessing that was a bug (before). This seems like the correct response from looking at https://postgrest.org/en/stable/api.html#singular-or-plural
